ursula_cli/ursula_inventory.py

```diff
@@ -0,0 +1,195 @@
+#!/usr/bin/env python
+# vim: tabstop=4 shiftwidth=4 softtabstop=4
+
+# Copyright 2016, Jesse Keating <jkeating@j2solutions.net>
+# All Rights Reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+
+import argparse
+import collections
+import json
+import logging
+import os
+import sys
+import yaml
+from ConfigParser import ConfigParser, NoOptionError, NoSectionError
+
+import ansible
+
+from ansible.constants import DEFAULTS, get_config, load_config_file
+from ansible.inventory import Inventory
+from ansible.inventory.dir import InventoryDirectory, get_file_parser
+from ansible.inventory.group import Group
+from ansible.parsing.dataloader import DataLoader
+from ansible.plugins import vars_loader
+from ansible.utils.vars import combine_vars
+from ansible.utils.vars import load_extra_vars
+from ansible.utils.vars import load_options_vars
+from ansible.vars import VariableManager
+
+LOG = logging.getLogger(__name__)
+
+
+def deep_update_dict(d, u):
+    for k, v in u.iteritems():
+        if isinstance(v, collections.Mapping):
+            r = deep_update_dict(d.get(k, {}) or {}, v)
```

Comment:
what's the `or` for?  potentially boolean false return value from `.get`?    I've written a similar function to accomplish a deep update of a dict, or a dict merge as I've called it.

https://gist.github.com/angstwad/bf22d1822c38a92ec0a9